src/main/scala/beam/agentsim/infrastructure/parking/ParkingZone.scala

```diff
@@ -116,18 +107,16 @@ object ParkingZone extends LazyLogging {
     * @param parkingZone the object to increment
     * @return True|False (representing success) wrapped in an effect type
     */
-  def claimStall(parkingZone: ParkingZone): Eval[Boolean] =
-    Eval.later {
-      if (parkingZone.parkingZoneId == DefaultParkingZoneId) {
-        // this zone does not exist in memory but it has infinitely many stalls to release
-        true
-      } else if (parkingZone.stallsAvailable - 1 >= 0) {
-        parkingZone.stallsAvailable -= 1
-        true
-      } else {
-        // log debug that we tried to claim a stall when there were no free stalls
-        false
-      }
+  def claimStall[GEO](parkingZone: ParkingZone[GEO]): Boolean =
+    if (parkingZone.parkingZoneId == DefaultParkingZoneId) {
```

Comment:
Why remove the `Eval.later`?

Reply:
I didn't find any usage of that eval except it executes right away after getting it.